src/main/scala-2.11/org/apache/james/gatling/smtp/SmtpAction.scala

```diff
@@ -1,70 +1,54 @@
+    Props(new SmtpAction(requestName, subject, body, statsEngine, next, protocol))
+}
+
+class SmtpAction(requestName: String,
                   subject: String,
                   body: String,
                   val statsEngine: StatsEngine,
                   val next: Action,
-                  protocol: SmtpProtocol) extends ChainableAction {
+                  protocol: SmtpProtocol) extends ChainableAction with Actor {
 
   val name = "sendMail"
 
   def execute(session: Session) {
-    val email: SimpleEmail = generateEmail(session)
-    val reqStart = System.currentTimeMillis()
-    val executionReport = sendMail(email)
-    val reqEnd = System.currentTimeMillis()
-
-    statsEngine.logResponse(session, requestName, ResponseTimings(reqStart, reqEnd), executionReport.status, None, executionReport.errorMessage)
+    def readSession(variableName: String): String = session.apply(variableName).as[String]
+    val request = SendMailRequest(session = session,
+      host = protocol.host,
+      port = protocol.port,
+      ssl = protocol.ssl,
+      from = readSession(UserFeeder.UsernameSessionParam),
+      to = readSession(UserFeeder.UsernameSessionParam),
+      subject = subject,
+      body = body,
+      credentials = readCredentials(readSession)(protocol))
+
+    context.actorOf(SmtpHandler.props()) ! request
 
     next ! session
   }
 
-  private def generateEmail(session: Session): SimpleEmail = {
-    def sessionVariable(variableName: String): String = {
-      session.apply(variableName).as[String]
-    }
-
-    val email = new SimpleEmail()
-    val userEmail = sessionVariable(UserFeeder.UsernameSessionParam)
-    if (protocol.auth) {
-      email.setAuthentication(sessionVariable(UserFeeder.UsernameSessionParam), sessionVariable(UserFeeder.PasswordSessionParam))
-    }
-    email.setHostName(protocol.host)
-    email.setSSLOnConnect(protocol.ssl)
-    email.setSmtpPort(protocol.port)
-    email.setSubject(subject + " [" + Math.random() + "]")
-    email.setMsg(body)
-    email.setFrom(userEmail)
-    email.addTo(userEmail)
-    email
+  override def receive: Receive = {
+    case session: Session => execute(session)
+    case executionReport: ExecutionReport =>
+      statsEngine.logResponse(executionReport.session, requestName, executionReport.responseTimings, executionReport.status, None, executionReport.errorMessage)
+    case Failure(e) =>
+      logger.error("Exception caught while sending mail", e)
   }
 
-  private def sendMail(email: SimpleEmail): ExecutionReport = {
-    Try(email.send()) match {
-      case Success(v) => GoodExecutionReport()
-      case Failure(e) =>
-        logger.error("Exception caught while sending mail", e)
-        BadExecutionReport(e.getMessage)
-    }
+  def readCredentials(sessionReader: String => String)(protocol: SmtpProtocol): Option[Credentials] = {
+    if (protocol.auth) None
```

Comment:
it's not what I expect readCredentials to do, maybe you could change method name to reflect that logic ?